src/blocks/battery.rs

```diff
@@ -66,107 +73,198 @@ impl ConfigBlock for Battery {
-        
-         else {
-            return Err(BlockError(
-                "battery".to_string(),
-                "Device does not support reading capacity, charge or energy".to_string(),
-            ));
-        }
 
-        let state = read_file("battery", &format!("{}status", self.device_path))?;
+            batteries_info.push(battery_summary);
+        } // done iterating over all batteries and collecting info
 
-        let energy_now = if file_exists(&format!("{}energy_now", self.device_path)) {
-            read_file("battery", &format!("{}energy_now", self.device_path))?
-                .parse::<u64>()
-                .block_error("battery", "failed to parse  energy_now")?
-        } else {
-            0
-        };
+        // We can now calculate the stats now.
+        // If a system has several batteries, we can calculate the full expected
+        // runtime of the device taking charge of all batteries into account
 
-        let energy_full = if file_exists(&format!("{}energy_full", self.device_path)) {
-            read_file("battery", &format!("{}energy_full", self.device_path))?
-                .parse::<u64>()
-                .block_error("battery", "failed to parse  energy_full")?
-        } else {
-            0
-        };
+        let mut total_current_charge = 0;
+        let mut total_energy_now = 0;
+        let mut total_energy_full = 0;
+        let mut total_max_charge = 0;
+        let mut total_power_now = 0;
+        let mut state: i64 = 0;
+        // use this to print percentages of each battery in the bar:
+        let mut percentages = String::new();
 
-        let power_now = if file_exists(&format!("{}power_now", self.device_path)) {
-            read_file("battery", &format!("{}power_now", self.device_path))?
-                .parse::<u64>()
-                .block_error("battery", "failed to parse current voltage")?
+        for battery in batteries_info {
+            total_current_charge += battery.current_charge;
+            total_energy_now += battery.energy_now;
+            total_energy_full += battery.energy_full;
+            total_max_charge += battery.max_charge;
+            total_power_now += battery.power_now;
+            // collect battery percentages
+            percentages.push_str(&format!("{}% ", battery.current_percentage));
+            // check if we have more batteries charging or discharging
+            // if state is negative, most batteries are discharging, positive if more are charging
+            if battery.state == "Charging" {
+                state += 1;
+            } else if battery.state == "Discharging" {
+                state -= 1;
+            } else if battery.state == "Unknown" {
+                // a battery may be Unknown when it's just sitting there not doing anything while the
+                // other one is being (dis/)charged
+                state += 0;
+            }
+        }
+
+        let device_state: DeviceState;
+        if state == 0 {
```

Comment:
Can we use `match` here instead? It would obviate the need for `unreachable!()`.

Reply:
done